frontend-web/webclient/app/Notifications/index.tsx

```diff
@@ -61,11 +61,17 @@ function resolveNotification(event: Notification): {
             }
             return {icon};
         case "JOB_COMPLETED":
-            const jobsCompletedTitle = event.meta.jobIds.length > 1 ?
-                `${event.meta.jobIds.length} jobs completed`
-                :
-                `${event.meta.appTitles[0]} completed`
-                ;
+            var jobsCompletedTitle: string;
+            
+            if (event.meta.jobIds.length > 1) {
+                jobsCompletedTitle = `${event.meta.jobIds.length} jobs completed`;
+            } else {
+                if (event.meta.jobNames[0]) {
+                    jobsCompletedTitle = `${event.meta.jobNames[0]} completed`;
+                } else {
+                    jobsCompletedTitle = `${event.meta.appTitles[0]} completed`;
+                }
+            }
```

Comment:
This could factored out into a function.
`const jobsCompletedTitle = extractNotificationText(event.meta, "completed");`